modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/application/HeadlessDiscoveryOpenAPIApplication.java

```diff
@@ -124,15 +185,59 @@ private void _addPaths(
 		}
 	}
 
+	private List<ApplicationDTO> _filterObjectsApplicationByCompany(
+		ApplicationDTO[] applicationDTOS) {
+
+		Stream<ApplicationDTO> applicationDTOStream = Arrays.stream(
```

Comment:
I am not sure if Brian will like the Stream...

What do you think about using the TransformUtil class?

```
private List<ApplicationDTO> _filterObjectsApplicationByCompany(
	ApplicationDTO[] applicationDTOS) {

	return TransformUtil.transformToList(
		applicationDTOS,
		applicationDTO -> {
			if (_objectApplicationCompanyIdsMap.containsKey(
					applicationDTO.base)) {

				List<String> companyIds =
					_objectApplicationCompanyIdsMap.get(
						applicationDTO.base);

				if (companyIds.contains(
						String.valueOf(
							CompanyThreadLocal.getCompanyId()))) {

					return applicationDTO;
				}

				return null;
			}

			return applicationDTO;
		});
}
```